include/rsg/host.hpp

```diff
@@ -0,0 +1,46 @@
+/* Copyright (c) 2015. The SimGrid Team. All rights reserved.              */
+
+/* This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Affero Licence (see in file LICENCE).        */
+
+#ifndef SRC_RSG_HOST_HPP_
+#define SRC_RSG_HOST_HPP_
+
+#include <boost/unordered_map.hpp>
+#include "rsg/engine.hpp"
+#include <xbt/string.hpp>
+
+namespace simgrid {
+  namespace rsg {
+
+    class Engine;
+
+    class Host {
+    	friend Engine;
+    private:
+    	Host(const char *name, unsigned long int remoteAddr);
+      Host(const std::string name, unsigned long int remoteAddr);
+
+    public:
+      simgrid::xbt::string const& name() const { return name_; }
+      double speed();
+      ~Host();
+
+    public:
+      static Host *by_name(std::string name);
```

Comment:
It would be better if your by_name() function could return a reference to the object instead of a pointer.
We discussed about it for the Actor module, and I changed the S4U interface to return references. I was hopping that you guys could do the same change in the RSG world, please.